pkg/middleware/stored_session.go

```diff
@@ -252,3 +262,40 @@ func (s *storedSessionLoader) validateSession(ctx context.Context, session *sess
 
 	return nil
 }
+
+// forceRefreshSession attempts to load a session as identified by the request cookies.
+// If no session is found, the request will be passed to the next handler.
+// If a session was loaded by a previous handler, it will not be refreshed.
+func (s *storedSessionLoader) forceRefreshSession(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		scope := middlewareapi.GetRequestScope(req)
+
+		// If refreshPeriod isn't defined then forward request to the next handler
+		if s.refreshPeriod <= time.Duration(0) {
+			next.ServeHTTP(rw, req)
+			return
+		}
+
+		session, err := s.store.Load(req)
+		if err == nil && session != nil {
+			err = s.refreshSessionIfNeeded(rw, req, session, true)
+		}
+
+		if err != nil {
```

Comment:
Generally we try to handle error cases first, leaving the "happy path" not needing to be wrapped in an if statement

Reply:
yep, i prefer the "return early pattern" as well.

it's not go but it illustrates the point nicely
https://gomakethings.com/the-early-return-pattern-in-javascript/

Reply:
@JoelSpeed / @tuunit 
It's been a busy year, but I finally got back around to this one.  I've made the changes you requested to have the code fail fast.  Please review again